apiserver/common/credentialcommon/modelcredential.go

```diff
@@ -16,12 +16,13 @@ import (
 	"github.com/juju/juju/environs"
 	environscloudspec "github.com/juju/juju/environs/cloudspec"
 	"github.com/juju/juju/environs/context"
+	"github.com/juju/juju/environs/instances"
 	"github.com/juju/juju/rpc/params"
 	"github.com/juju/juju/state"
 )
 
 // ValidateExistingModelCredential checks if the cloud credential that a given model uses is valid for it.
-func ValidateExistingModelCredential(backend PersistentBackend, callCtx context.ProviderCallContext, checkCloudInstances bool) (params.ErrorResults, error) {
+func ValidateExistingModelCredential(backend PersistentBackend, callCtx context.ProviderCallContext, checkCloudInstances bool, modelMigrationCheck bool) (params.ErrorResults, error) {
```

Comment:
You can actually remove the `checkCloudInstances` argument, because it is supplied based on whether we're migrating or not, which you now have another argument for. You can remove it.

Reply:
There's a little subtlety here, removing that would actually change semantics. I also wrote about this in the comments I just added after your review. 

Normally it's ok to have more instances than machines, i.e. if the `checkCloudInstances = false`, this function will only check if there's an instance for every machine (i.e. extra instances are fine). However, if the `checkCloudInstances = true`, then it will expect an exact 1:1 mapping between instances and machines to deem the credential as valid. So removing that parameter would mean we'd lock that check semantics to one way or another.

Reply:
Yes I see; it's `cloud.Type != manual` in the migration check.